src/submission/validation-clinical/therapy.ts

```diff
@@ -17,42 +17,234 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+import { DeepReadonly } from 'deep-freeze';
+import _ from 'lodash';
+import * as utils from './utils';
 import {
   SubmissionValidationError,
   SubmittedClinicalRecord,
   DataValidationErrors,
   SubmissionValidationOutput,
+  ClinicalSubmissionRecordsByDonorIdMap,
 } from '../submission-entities';
 import {
   ClinicalEntitySchemaNames,
-  TreatmentFieldsEnum,
   ClinicalTherapyType,
+  RadiationFieldsEnum,
+  TreatmentFieldsEnum,
 } from '../../common-model/entities';
-import { DeepReadonly } from 'deep-freeze';
-import { Donor, Treatment } from '../../clinical/clinical-entities';
-import * as utils from './utils';
-import _ from 'lodash';
 import { getSingleClinicalObjectFromDonor } from '../../common-model/functions';
+import { donorDao } from '../../clinical/donor-repo';
+import { ClinicalInfo, Donor, Treatment } from '../../clinical/clinical-entities';
+import { ClinicalQuery } from '../../clinical/clinical-api';
 import { isValueEqual } from '../../utils';
 
 export const validate = async (
   therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
   existentDonor: DeepReadonly<Donor>,
   mergedDonor: Donor,
+  submittedRecords: DeepReadonly<ClinicalSubmissionRecordsByDonorIdMap>,
 ): Promise<SubmissionValidationOutput> => {
   // ***Basic pre-check (to prevent execution if missing required variables)***
   if (!therapyRecord || !mergedDonor || !existentDonor) {
     throw new Error("Can't call this function without a registerd donor & therapy record");
   }
 
-  const errors: SubmissionValidationError[] = [];
+  let errors: SubmissionValidationError[] = [];
   const treatment = getTreatment(therapyRecord, mergedDonor, errors);
   if (!treatment) return { errors };
-  checkTreatementHasCorrectTypeForTherapy(therapyRecord, treatment, errors);
+  // TODO: Refactor use of push
+  checkTreatmentHasCorrectTypeForTherapy(therapyRecord, treatment, errors);
+
+  const crossFileErrors = await crossFileValidator(
+    mergedDonor,
+    treatment,
+    therapyRecord,
+    submittedRecords,
+  );
+
+  errors = [...errors, ...crossFileErrors];
   return { errors };
 };
 
-export function checkTreatementHasCorrectTypeForTherapy(
+const crossFileValidator = async (
+  donor: Donor,
+  treatment: DeepReadonly<Treatment>,
+  therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
+  // submitted records needed to validate current submission
+  submittedRecords: DeepReadonly<ClinicalSubmissionRecordsByDonorIdMap>,
+) => {
+  const isRadiationRecord = Object.values(RadiationFieldsEnum).some(field =>
+    Object.keys(therapyRecord).includes(field),
+  );
+
+  const radiationErrors: SubmissionValidationError[] = isRadiationRecord
+    ? await validateRadiationRecords(donor, treatment, therapyRecord, submittedRecords)
+    : [];
+
+  const crossFileErrors = [...radiationErrors];
```

Comment:
The idea here is that it is only equal to `radiationErrors` until we add more validations
I try not to spend much time solving 'future problems' that might not exist, but my next ~3 tickets are all cross-file validation
So this will look more like:
`const crossFileErrors = [...radiationErrors, ...followUpErrors, ...chemotherapyErrors];`
Given that perspective, let me know if there's a better approach

Reply:
Great! Yes this just didn't click for me on reading - makes sense as is!